src/hw-monitor.cpp

```diff
@@ -174,6 +174,17 @@ namespace librealsense
             newCommand.receivedCommandData + newCommand.receivedCommandDataLength);
     }
 
+    std::vector<uint8_t> hw_monitor::build_raw_data(const uint32_t opcode, const uint32_t param1, const uint32_t param2,
+        const uint32_t param3, const uint32_t param4, const std::vector<uint8_t>& data) const
+    {
+        int length;
+        std::vector<uint8_t> result;
+        result.resize(IVCAM_MONITOR_MAX_BUFFER_SIZE);
+        fill_usb_buffer(opcode, param1, param2, param3, param4, data.data(), (uint32_t)data.size(), result.data(), length);
```

Comment:
`fill_usb_buffer` (and `hwmon_cmd`) takes integers (`int`) for the parameters... so why do we accept unsigned values?

Reply:
opcode and all params are 4 byte (32 bit) each.
I just though that this is the best way to describe that each parameter is 4 bytes.
in fill_usb_buffer, we are aligning these int inputs you are talking about into 32 bits each (with memcpy)
so I actually see that the problem is in fill_usb_buffer.. which accepts int params, while in in sending and receiving data its more about "bytes"...
anyway.. we will talk about this tomorrow.